src/main/java/controller/Controller.java

```diff
@@ -1,38 +1,59 @@
 package controller;
 
 import annotation.GetMapping;
+import annotation.PostMapping;
 import db.Database;
+import db.SessionManager;
 import http.HttpResponse;
 import model.User;
+import model.UserFactory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Map;
+import java.util.Optional;
 
+/**
+ * "redirect:" 를 반환하는 경우 index.html로 이동한다.
+ */
 public class Controller {
     private static final Logger logger = LoggerFactory.getLogger(Controller.class);
     private final String USER_ID = "userId";
     private final String PASSWORD = "password";
-    private final String NAME = "name";
-    private final String EMAIL = "email";
+    private final String SID = "sid";
 
-    // GET 회원가입
-    @GetMapping(value = "/user/create")
-    public String createUser(Map<String, String> query) {
-        logger.debug("Received GET request to create user.");
-
-        String userId = query.get(USER_ID);
-        String password = query.get(PASSWORD);
-        String name = query.get(NAME);
-        String email = query.get(EMAIL);
-
-        logger.debug("받은 유저 데이터 - userId: {}, password: {}, name: {}, email: {}", userId, password, name, email);
+    //회원가입
+    @PostMapping(value = "/user/create")
+    public HttpResponse createUser(Map<String, String> body) {
+        User user = UserFactory.createUserFrom(body);
+        Database.addUser(user);
+        logger.debug("user 생성 : {}", Database.findUserById(body.get(USER_ID)));
 
-        User user = new User(userId, password, name, email);
+        return HttpResponse.redirect();
+    }
 
-        Database.addUser(user);
-        logger.debug("User created: {}", user);
+    //로그인
+    @PostMapping(value = "/user/login")
+    public HttpResponse login(Map<String, String> body) {
+        Optional<User> user = Database.findUserById(body.get(USER_ID));
+        if (user.isEmpty() || !user.get().checkPassword(body.get(PASSWORD))) {
+            logger.error("아이디 혹은 비밀번호가 일치하지 않습니다");
+            return HttpResponse.ok("/user/login_failed.html");
+        }
+
+        String sid = SessionManager.createUserSession(user.get());
```

Comment:
지금도 충분히 괜찮지만 개인적으로는 단어의 중복을 피하려고 하는 편인데요.
```suggestion
        String sid = SessionManager.create(user);
```
저라면 요렇게?